### ISSUE TITLE
Exception in Astroid with multiple inheritance and properties

### ISSUE THREAD

**Issue by dcbaker** [PyCQA/pylint#840]
pylint: 1.5.4
astroid: 1.4.4
python: 2.7.11, 3.3.6, 3.5.1

When using a class hierarchy at least three deep with property decorators, and overriding those decorators and exception is raised in astroid.

The exception:
```txt
Traceback (most recent call last):
  File "/home/user/.pyenv/versions/python-3.3/lib/python3.3/site-packages/astroid/decorators.py", line 34, in cached
    return cache[func]
KeyError: <bound method FunctionDef.decoratornames of <FunctionDef(foo) l.13 [test] at ...>
```

The most trivial example I could come up with below:
```python
class Parent(object):
    def __init__(self):
        self.__foo = ['foo']

    @property
    def foo(self):
        return self.__foo

class Child(Parent):
    @Parent.foo.getter
    def foo(self):
        return super(Child, self).foo + ['bar']

class Child2(Child):
    @Child.foo.getter
    def foo(self):
        return super(Child2, self).foo + ['oink']
```

Simply running pylint on this chunk will reproduce the error.

**Comment by PCManticore** [comment]
I have a totally different stacktrace and an error when running this with 1.5.4:

 File "d:\projects\repos\pylint\pylint\checkers\typecheck.py", line 371, in visit_attribute
   if not [n for n in owner.getattr(node.attrname)
 File "d:\projects\repos\astroid\astroid\objects.py", line 189, in getattr
   return list(self.igetattr(name, context=context))
 File "d:\projects\repos\astroid\astroid\objects.py", line 176, in igetattr
   elif bases._is_property(inferred):
 File "d:\projects\repos\astroid\astroid\bases.py", line 53, in _is_property
   if PROPERTIES.intersection(meth.decoratornames()):
 File "C:\Users\Claudiu\AppData\Local\Programs\Python\Python35-32\lib\site-packages\wrapt\wrappers.py", line 561, in __call__
   args, kwargs)
 File "d:\projects\repos\astroid\astroid\decorators.py", line 42, in cached
   cache[func] = result = func(*args, **kwargs)
 File "d:\projects\repos\astroid\astroid\scoped_nodes.py", line 905, in decoratornames
   for infnode in decnode.infer():
 File "d:\projects\repos\astroid\astroid\decorators.py", line 99, in wrapped
   res = next(generator)
 File "d:\projects\repos\astroid\astroid\decorators.py", line 157, in raise_if_nothing_inferred
   raise exceptions.InferenceError(**error.args[0])
stroid.exceptions.InferenceError: Inference failed for <Attribute.getter l.10 at 0x3a20050>.

This is probably something that's easily fixable, since the exception should leak out like this.

**Comment by dcbaker** [comment]
I get something like that too, as a secondary trace on 3.x

**Comment by PCManticore** [comment]
This should be fixed in astroid (https://github.com/PyCQA/astroid/commit/7278409ee6e578c1f40204b062def17d704a08e6). Let me know if it works for you. My plan is to release a new bug fix release of pylint (1.5.5) tomorrow or in the weekend.
